packages/gestalt/src/Touchable.js

```diff
@@ -73,126 +77,167 @@ const getRoundingStyle = (r: Rounding): Style => {
+  onFocus,
+  onMouseEnter,
+  onMouseLeave,
+  onTouch,
+  pressBackgroundEnabled = false,
+  pressScaleEnabled = false,
+  rounding = 0,
+}: Props) {
+  const [isFocused, setFocused] = React.useState<boolean>(false);
+  const [isPressed, setPressed] = React.useState<boolean>(false);
+  const [coordinate, setCoordinate] = React.useState<{|
+    x: number,
+    y: number,
+  |}>({ x: 0, y: 0 });
+  const ignoreNextFocusEventRef = React.useRef<boolean>(false);
+
+  const classes = classnames(
+    styles.touchable,
+    toProps(getRoundingStyle(rounding)).className,
+    {
+      [styles.fullHeight]: fullHeight,
+      [styles.fullWidth]: fullWidth,
+      [styles[mouseCursor]]: !disabled,
+      [styles.pressScale]: !disabled && pressScaleEnabled && isPressed,
+      [styles.pressBackground]:
+        !disabled && pressBackgroundEnabled && (isPressed || isFocused),
     }
-  };
-
-  render() {
-    const {
-      accessibilityControls,
-      accessibilityExpanded,
-      accessibilityHaspopup,
-      accessibilityLabel,
-      children,
-      disabled = false,
-      fullWidth = true,
-      fullHeight,
-      mouseCursor = 'pointer',
-      rounding = 0,
-    } = this.props;
-
-    const classes = classnames(
-      styles.touchable,
-      toProps(getRoundingStyle(rounding)).className,
-      {
-        [styles.fullHeight]: fullHeight,
-        [styles.fullWidth]: fullWidth,
-        [styles[mouseCursor]]: !disabled,
-      }
-    );
-
-    return (
-      <div
-        aria-controls={accessibilityControls}
-        aria-disabled={disabled}
-        aria-expanded={accessibilityExpanded}
-        aria-haspopup={accessibilityHaspopup}
-        aria-label={accessibilityLabel}
-        className={classes}
-        onClick={this.handleClick}
-        onBlur={this.handleBlur}
-        onFocus={this.handleFocus}
-        onMouseEnter={this.handleMouseEnter}
-        onMouseLeave={this.handleMouseLeave}
-        onKeyPress={this.handleKeyPress}
-        role="button"
-        tabIndex={disabled ? null : '0'}
-      >
-        {children}
-      </div>
-    );
-  }
+  );
+
+  return (
+    <div
+      aria-controls={accessibilityControls}
+      aria-disabled={!!disabled}
+      aria-expanded={accessibilityExpanded}
+      aria-haspopup={accessibilityHaspopup}
+      aria-label={accessibilityLabel}
+      className={classes}
+      onContextMenu={event => event.preventDefault()}
+      onClick={event => {
+        if (!disabled && onTouch) {
+          onTouch({ event });
+        }
+      }}
+      onBlur={event => {
+        if (!disabled && onBlur) {
+          onBlur({ event });
+        }
+        setFocused(false);
+        setPressed(false);
+      }}
+      onFocus={event => {
+        if (!disabled && onFocus) {
+          onFocus({ event });
+        }
+        if (!isPressed && !ignoreNextFocusEventRef.current) {
+          setFocused(true);
+        }
+        ignoreNextFocusEventRef.current = false;
+        event.stopPropagation();
```

Comment:
calling `event.stopPropagation();` in `onFocus` is a breaking change here. See https://phabricator.pinadmin.com/differential/changeset/?ref=7777595